scripts/jmwalletd.py

```diff
@@ -0,0 +1,626 @@
+                        walletname=self.wallet_service.get_wallet_name(),
+                        already_loaded=False,token=encoded_token)
+
+    @app.route('/wallet/<string:walletname>/unlock', methods=['POST'])
+    def unlockwallet(self, request, walletname):
+        print_req(request)
+        #print(get_current_chain_params())
+        assert isinstance(request.content, BytesIO)
+        auth_json = self.get_POST_body(request, ["password"])
+        if not auth_json:
+            raise InvalidRequestFormat()
+        password = auth_json["password"]
+        if self.wallet_service is None:
+            wallet_path = get_wallet_path(walletname, None)
+            try:
+                wallet = open_test_wallet_maybe(
+                        wallet_path, walletname, 4,
+                        password=password.encode("utf-8"),
+                        ask_for_password=False)
+            except StoragePasswordError:
+                raise NotAuthorized("invalid password")
+            except StorageError as e:
+                # e.g. .lock file exists:
+                raise NotAuthorized(repr(e))
+            return self.initialize_wallet_service(request, wallet)
+        else:
+            print('wallet was already unlocked.')
+            return response(request,
+                            walletname=self.wallet_service.get_wallet_name(),
+                            already_loaded=True)
+
+
+    #This route should return list of current wallets created.
+    @app.route('/wallet/all', methods=['GET'])
+    def listwallets(self, request):
+        #this is according to the assumption that wallets are there in /.joinmarket by default, also currently path for linux system only.
+        #first user taken for path
+        user_path = glob.glob('/home/*/')[0]
+        
+        wallet_dir = f"{user_path}.joinmarket/wallets/*.jmdat"
+        wallets = (glob.glob(wallet_dir))
+        
+        offset = len(user_path)+len('.joinmarket/wallets/')
+        #to get only names
+        short_wallets = [wallet[offset:] for wallet in wallets]
+        return response(request,wallets=short_wallets)
+
+    #route to get external address for deposit
+    @app.route('/address/new/<string:mixdepth>',methods=['GET'])
+    def getaddress(self, request, mixdepth):
+        # self.check_cookie(request)
+        if not self.wallet_service:
+            raise NoWalletFound()
+        mixdepth = int(mixdepth)
+        address = self.wallet_service.get_external_addr(mixdepth)
+        return response(request,address=address)
+
+    #route to list utxos
+    @app.route('/wallet/transactions',methods=['GET'])
+    def listUtxos(self, request):
+        if not self.wallet_service:
+            raise NoWalletFound()
+        utxos = wallet_showutxos(self.wallet_service, False)
+        
+        return response(request,transactions=utxos)
+
+    #return True for now
+    def filter_orders_callback(orderfees, cjamount,dummy):
```

Comment:
Same comment here as for `taker_finished`, i.e. it's an instance method and first arg should be `self`. Probably that's why `dummy`?

But reading this just makes me realize, we could and should just have the user accept the orders (offers, sorry name confusion, think of them as the same thing), as long as it's in the fees bound they've set. Having an extra "is this really OK?" is just a luxury, as long as they've decided upfront with their `max_cj_fee_rel` and `max_cj_fee_abs` then they don't *need* to be interrupted at this point. I mean, `return True` is a valid starting point for tests :smile:

Reply:
Well, actually it's the `max_cj_fee = (1,float('inf'))` that represents the "valid starting point for tests", not the `return True` here.... Just fill in the right fee settings from the config, I think.

Which makes me realize a good version of this api has to allow editing the config, or at least some of it.